scripts/language_model/run_glue.py

```diff
@@ -43,75 +44,148 @@
-parser.add_argument('--task_name', default='MRPC', type=str,
-                    help='The name of the task to fine-tune.')
-
-parser.add_argument('--model_name', type=str, default='xlnet_cased_l12_h768_a12',
-                    choices=['xlnet_cased_l24_h1024_a16', 'xlnet_cased_l12_h768_a12'],
-                    help='The name of pre-trained XLNet model to fine-tune')
-
-parser.add_argument('--dataset', type=str, default='126gb',
-                    help='The dataset BERT pre-trained with.')
+    '--only_inference',
+    action='store_true',
+    help=
+    'If set, we skip training and only perform inference on dev and test data.'
+)
 
-parser.add_argument('--output_dir', type=str, default='./output_dir',
-                    help='The output directory where the model params will be written.')
+# Initializing config
+parser.add_argument('--seed', type=int, default=2, help='Random seed')
 
+# I/O config
 parser.add_argument(
-    '--only_inference', action='store_true',
-    help='If set, we skip training and only perform inference on dev and test data.')
-
+    '--output_dir',
+    type=str,
+    default='./output_dir',
+    help='The output directory where the model params will be written.')
 parser.add_argument(
-    '--model_parameters', type=str, default=None,
+    '--model_parameters',
+    type=str,
+    default=None,
     help='A parameter file for the model that is loaded into the model'
     ' before training/inference. It is different from the parameter'
     ' file written after the model is trained.')
 
-parser.add_argument(
-    '--early_stop', type=int, default=None,
-    help='Whether to perform early stopping based on the metric on dev set. '
-    'The provided value is the patience. ')
-
-parser.add_argument('--dropout', type=float, default=0.1, help='dropout')
-parser.add_argument('--attention_dropout', type=float, default=0.1, help='attention dropout')
-parser.add_argument('--lr_decay', type=str, default='linear', help='lr decay')
 args = parser.parse_args()
 
 
-def split_and_load(arrs, ctx):
+def split_array(arr, num_of_splits):
+    """split an array into equal pieces"""
+    size = arr.shape[0]
+    if size < num_of_splits:
+        return [arr[i:i + 1] for i in range(size)]
+    slice_len, rest = divmod(size, num_of_splits)
+    div_points = [0] + [(slice_len * index + min(index, rest) + slice_len +
+                         (index < rest)) for index in range(num_of_splits)]
+    slices = [
+        arr[div_points[i]:div_points[i + 1]] for i in range(num_of_splits)
+    ]
+    return slices
+
+
+def split_and_load(arrs, _ctxs):
     """split and load arrays to a list of contexts"""
     assert isinstance(arrs, (list, tuple))
     # split and load
-    loaded_arrs = [mx.gluon.utils.split_and_load(arr, ctx, even_split=False) for arr in arrs]
+    loaded_arrs = [[
+        i.as_in_context(ctx)
+        for i, ctx in zip(split_array(arr, len(_ctxs)), _ctxs)
```

Comment:
This is a workaround for MXNet 1.6, as https://github.com/apache/incubator-mxnet/pull/17123 is only available in MXNet 1.7?

Could you add a comment like "TODO Replace with ... once targeting MXNet 1.7"